### PR TITLE
BLOCKCHAIN-240 + BLOCKCHAIN-246 + BLOCKCHAIN-247 - optimize company data format + small UX improvements + bugfixes

### PR DESCRIPTION
**Pull request by kacperzuk-neti**
Using sample data:

Original JSON encoding: 3279 bytes
JSON encoding + deflate compressed: 827 bytes

SCALE encoded: 1073 bytes
This PR - SCALE encoded + deflate compressed: 459 bytes

So for sample data we've reduced data size by 86%.

### CODE REVIEW

File: src/api/nodeRpcCall.js [hunk 1]
```diff
@@ -913,52 +1012,60 @@ const getOfficialUserRegistryEntries = async (walletAddress) => {
...
-                pushObject['isEncrypted'] = dynamicFieldDataArray[keyIsEncrypted]
-                crossReferencedFieldDataArray.push(JSON.parse(JSON.stringify(pushObject)))
-              }
+    let companyData;
+    try {
+      const compressed = companyRegistryEntity.unwrap().data;
+      companyData = api.createType('CompanyData', pako.inflate(compressed)).toJSON();
+    } catch(e) {
+      console.error("Invalid company data", e); // FIXME actually display this to user somehow?
+      return;
```
Comment (MichalLustykNeti) [inline src/api/nodeRpcCall.js:1021]: Shouldn't we return null?

File: src/api/nodeRpcCall.js [hunk 2]
```diff
@@ -1088,20 +1168,7 @@ const getLandNFTMetadataJson = async (collection_id, nft_id) => {
...
-      },
-      "LandMetadata": {
-        "demarcation": "BoundedVec<Coords, u32>",
-        "type": "Text",
-        "status": "Text",
-      }
-    }
-  });
+  const api = await getApi();
   /*let metadata = {
```
Comment (MichalLustykNeti) [inline src/api/nodeRpcCall.js:1172]: i believe we should clean up this

File: src/api/nodeRpcCall.js (hunk 1 above)
Comment (kacperzuk-neti) [reply src/api/nodeRpcCall.js:1021]: Not sure it matters.

1. It should never happen
2. Both `undefined` (which I think will result in infinite loader) and `null` (which I think will result in "No registered companies") are wrong. I kinda thing `null` is worse, as it just lies to user, while `undefined` at least will clearly show as something that's broken. We just need a proper error handling for this.

I've created BLOCKCHAIN-254 ticket to implement the error handling. I don't want to modify this file, as there are other PRs that already rely on it, so I want to limit changes to critical only.

File: src/api/nodeRpcCall.js (hunk 2 above)
Comment (kacperzuk-neti) [reply src/api/nodeRpcCall.js:1172]: Out of scope for this PR. And there are other PRs that depend on this one, so such change will cause more trouble than it's worth now.

File: src/components/Registries/RegistriesCompanies/CompanyCRUD/index.js [hunk 3]
```diff
@@ -1,38 +1,50 @@
...
-    if(registryCRUDAction.action == 'edit'){
-      formObject = registryCRUDAction.dataObject
-      buttonMessage = 'Submit Company Edit'
-      companyId = registryCRUDAction.dataObject.id
-      callback = ((companyData) =>{})
+    if (registryCRUDAction.action === 'edit') {
+      formObject = registryCRUDAction.dataObject;
+      buttonMessage = 'Submit Company Edit';
+      companyId = registryCRUDAction.dataObject.id;
+      callback = ((_companyData) => {});
```
Comment (DorianSternVukotic) [inline src/components/Registries/RegistriesCompanies/CompanyCRUD/index.js:39]: why the underscore ?

Comment (kacperzuk-neti) [reply src/components/Registries/RegistriesCompanies/CompanyCRUD/index.js:39]: We're trying to make new/modified code have no errors when linted with eslint. One of the lints is that unused variables should be prefixed with `_`. `_companyData` here is unused.

**Comment by kacperzuk-neti** [comment]
Approved on dev chat.

**Pull request merged**